crates/ra_hir_ty/src/traits/chalk.rs

```diff
@@ -17,6 +17,8 @@ use crate::{
     ProjectionTy, Substs, TraitRef, Ty, TypeCtor,
 };
 
+pub(super) mod tls;
+
 #[derive(Debug, Copy, Clone, Hash, PartialOrd, Ord, PartialEq, Eq)]
 pub struct Interner;
```

Comment:
Just a wild supposition (not very familiar with `ra_hir_ty` codebase), but it looks like the `Interner` struct may want to hold a reference to `&dyn HirDatabase` to avoid the hacks with tls...

Reply:
The debug methods don't get the Interner, so this wouldn't help.